### PR TITLE
Pubsub func testing

### PR DESCRIPTION
**Pull request by ludwikbukowski**
- Fixed integration in deps with rebar3
- publish_entry_children matches correctly on xmlel list

### CODE REVIEW

File: src/escalus_pubsub_stanza.erl [hunk 1]
```diff
@@ -258,3 +259,21 @@ get_subscription_change_list_stanza(SubscriptionChangeData) ->
...
+publish_content_test(Config) ->
+  Title = <<"Title">>,
+  Data = <<"Some data">>,
+  PublishItemId = <<"publishid">>,
+  DestinationNode = <<"localhost.node">>,
+  User = <<"alice@localhost">>,
+  Entry = escalus_pubsub_stanza:publish_entry(prepare_body_for_publish(Title, Data)),
+  Content = escalus_pubsub_stanza:publish_item(PublishItemId, Entry),
+  PublishToNodeIq = escalus_pubsub_stanza:publish_node_with_content_stanza(DestinationNode, Content),
+  Stanza = escalus_pubsub_stanza:iq_with_id(set, PublishItemId, DestinationNode, User, [PublishToNodeIq]).
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 273 is too long:   Stanza = escalus_pubsub_stanza:iq_with_id(set, PublishItemId, DestinationNode, User, [PublishToNodeIq])..

File: src/escalus_pubsub_stanza.erl [hunk 2]
```diff
@@ -66,7 +64,7 @@ publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItem
                         _ ->
                             create_publish_node_content_stanza(DestinationTopicName, PublishItemId)
                     end,
-    IqId = <<"publish1">>,
+    IqId = ?SAMPLE_ID,
     escalus_pubsub_stanza:iq_with_id(set, IqId, DestinationNode, User,  [PublishToNode]).
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 68 is too long:     escalus_pubsub_stanza:iq_with_id(set, IqId, DestinationNode, User,  [PublishToNode])..

File: src/escalus_pubsub_stanza.erl [hunk 3]
```diff
@@ -66,7 +64,7 @@ publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItem
                         _ ->
                             create_publish_node_content_stanza(DestinationTopicName, PublishItemId)
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 65 is too long:                             create_publish_node_content_stanza(DestinationTopicName, PublishItemId).

File: test/escalus_stanza_SUITE.erl [hunk 4]
```diff
@@ -62,3 +62,5 @@ numbers_as_arguments(_) ->
     ?eq(Example,
         M:from_template("<el int='{{int}}' pi='{{pi}}'/>",
```
Comment (gadgetci) [inline test/escalus_stanza_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 63. Only modules that define callbacks should make dynamic calls.

File: src/escalus_pubsub_stanza.erl [hunk 5]
```diff
@@ -87,7 +85,7 @@ iq_set_get_rest(SrcIq, Id, From) ->
 
 %% ----------------------------- sample entry bodies ------------------------
 
-entry_body_sample1() ->
+entry_body_sample() ->
     [
      #xmlel{name = <<"title">>, children  = [ #xmlcdata{content=[<<"The title of content.">>]}]},
      #xmlel{name = <<"summary">>, children = [ #xmlcdata{content=[<<"To be or not to be...">>]}]}
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 91 is too long:      #xmlel{name = <<"summary">>, children = [ #xmlcdata{content=[<<"To be or not to be...">>]}]}.

File: src/escalus_pubsub_stanza.erl [hunk 6]
```diff
@@ -258,3 +259,21 @@ get_subscription_change_list_stanza(SubscriptionChangeData) ->
...
+
+publish_content_test(Config) ->
+  Title = <<"Title">>,
+  Data = <<"Some data">>,
+  PublishItemId = <<"publishid">>,
+  DestinationNode = <<"localhost.node">>,
+  User = <<"alice@localhost">>,
+  Entry = escalus_pubsub_stanza:publish_entry(prepare_body_for_publish(Title, Data)),
+  Content = escalus_pubsub_stanza:publish_item(PublishItemId, Entry),
+  PublishToNodeIq = escalus_pubsub_stanza:publish_node_with_content_stanza(DestinationNode, Content),
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 272 is too long:   PublishToNodeIq = escalus_pubsub_stanza:publish_node_with_content_stanza(DestinationNode, Content),.

File: src/escalus_pubsub_stanza.erl [hunk 7]
```diff
@@ -258,3 +259,21 @@ get_subscription_change_list_stanza(SubscriptionChangeData) ->
...
       SubscriptionChangeData).
+
+
+publish_content_test(Config) ->
+  Title = <<"Title">>,
+  Data = <<"Some data">>,
+  PublishItemId = <<"publishid">>,
+  DestinationNode = <<"localhost.node">>,
+  User = <<"alice@localhost">>,
+  Entry = escalus_pubsub_stanza:publish_entry(prepare_body_for_publish(Title, Data)),
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 270 is too long:   Entry = escalus_pubsub_stanza:publish_entry(prepare_body_for_publish(Title, Data)),.

File: src/escalus_pubsub_stanza.erl [hunk 8]
```diff
@@ -87,7 +85,7 @@ iq_set_get_rest(SrcIq, Id, From) ->
 
 %% ----------------------------- sample entry bodies ------------------------
 
-entry_body_sample1() ->
+entry_body_sample() ->
     [
      #xmlel{name = <<"title">>, children  = [ #xmlcdata{content=[<<"The title of content.">>]}]},
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 90 is too long:      #xmlel{name = <<"title">>, children  = [ #xmlcdata{content=[<<"The title of content.">>]}]},.

File: src/escalus_pubsub_stanza.erl [hunk 9]
```diff
@@ -14,36 +14,34 @@
...
+  publish_entry/1,
+  pubsub_stanza/2,
+  publish_node_with_content_stanza/2,
+  publish_sample_content_stanza/5,
+  retract_from_node_stanza/2,
+  retrieve_subscriptions_stanza/1,
+  set_subscriptions_stanza/2,
+  subscribe_by_user_stanza/4,
+  unsubscribe_by_user_stanza/4
+  , publish_content_test/1]).
```
Comment (lavrin) [inline src/escalus_pubsub_stanza.erl]: ;)

File: src/escalus_pubsub_stanza.erl [hunk 10]
```diff
@@ -114,11 +112,14 @@ entry_body_with_sample_device_id_2() ->
 %% provide EntryBody as list of anything compliant with exml entity records.
 
 publish_entry_children([]) ->
-    entry_body_sample1();
+    entry_body_sample();
 
-publish_entry_children([#xmlel{}] = EntryBody) ->
+publish_entry_children([#xmlel{} | []] = EntryBody) ->
```
Comment (lavrin) [inline src/escalus_pubsub_stanza.erl]: This pattern is overly complex - it's _obvious already_ that we match on a one element long list, there's no point in adding `| []` - it only adds line noise.

**Pull request closed without merging**